src/plugins/stdioproc/README.md

```diff
@@ -0,0 +1,201 @@
+```
+
+During `elektraStdprocioOpen` the plugin will collect the `args/#` and `env/#` values into two arrays `argv` and `envp`.
+It then `fork`s a new process and the child calls `execve` with the path from `app` as well as `argv` and` envp`.
+The child process is expected to listen to `stdin` and write to `stdout` according to the protocol described below.
+
+If communication can be established, the child-process will be kept running until `elektraStdprocioClose`.
+
+## Protocol
+
+The entire protocol is text-based (apart from binary key values) and request-response-based, and happens over `stdin`/`stdout`.
+The parent process sends request to the child process.
+The child then processes the request and sends a response back.
+
+To encode keysets and keys we use the format of the `dump` plugin.
+It is important to note that the `dump` plugin is instructed to write the full keynames (normally it removes the parent prefix).
+There is, however, an exception during initialization, which is described in the appropriate section.
+The child-process may also delegate directly to the `dump` plugin, or re-implement the encoding.
+
+The communications protocol used by the plugin has 3 phases:
+
+- Initialization: Initial handshake after first starting the child-process.
+- Operation: The main phase, which implements all the operations of an Elektra plugin (open, get, set, close, ...).
+- Termination: The plugin is being closed and the parent-process tells the child to shut down.
+
+> **Note:** In the sections below we use the following format to describe messages:
+>
+> ```
+> Parent > Child
+>
+> HELLO WORLD
+> [users]
+> (user)
+> ```
+>
+> This denotes a message sent from parent to child, containing
+>
+> - the literal text `HELLO WORLD`
+> - followed by a newline
+> - followed by a keyset called `users`
+> - followed by a newline
+> - followed by dynamic text called `user`
+>
+> The names `users` and `user` don't actually appear in the message.
+> They are only used as a reference for the descriptions of the message.
+>
+> The actual message sent could look like this:
+>
+> ```
+> HELLO WORLD
+> kdbOpen 2
+> $key string 3 5
+> joe
+> 12345
+> $end
+> appleseed
+> ```
+
+### Initialization
+
+The parent writes the protocol header to `stdin` of the child.
+
+```
+Parent > Child
+
+ELEKTRA_STDIOPROC INIT v1
+```
+
+The child must respond with an acknowledgement, followed by a `contract` keyset.
+
+```
+Child > Parent
+
+ELEKTRA_STDIOPROC ACK v1
+(name)
+[contract]
+```
+
+Here `(name)` is the module name the child uses.
+This will be used to replace `stdioproc` in the contract of the plugin.
+The `[contract]` keyset must contain the `infos` keys, which will be used replace the ones at the top of this README and the ones in `stdioproc.c`.
+It must also contain `exports/_` keys for every operation that is implemented by the child.
+All keys in `[contract]` should be below `system:/elektra/modules/stdioproc`.
```

Comment:
Maybe allow any `system:/elektra/modules/<modulename>`?

Reply:
IMO this is only an option, if `<modulename>` must match the previously sent `(name)`. This would also create a new source of errors.